src/awesome/awesome.rs

```diff
@@ -248,6 +249,19 @@ fn load_image<'lua>(lua: &'lua Lua, file_path: String) -> rlua::Result<Value<'lu
     rlua::LightUserData(surface_ptr as _).to_lua(lua)
 }
 
+/// Convert a pixbuf to a cairo image surface.
+/// Returns either a cairo surface as light user data, nil and an error message
+fn pixbuf_to_surface<'lua>(lua: &'lua Lua, pixbuf: LightUserData) -> rlua::Result<Value<'lua>> {
```

Comment:
The raw pointer being passed in here from Lua with no checks possible for validity makes me sad. But there's nothing we can do I suppose.

Perhaps once Way Cooler is stable we can look into making that safer.